src/AST-Core/ASTCacheMissStrategy.class.st

```diff
@@ -6,5 +6,8 @@ Class {
 
 { #category : #accessing }
 ASTCacheMissStrategy >> getASTFor: aCompiledMethod [
-	^ aCompiledMethod parseTree doSemanticAnalysisIn: aCompiledMethod methodClass
+	| ast |
+	ast := aCompiledMethod parseTree doSemanticAnalysisIn: aCompiledMethod methodClass.
```

Comment:
it means that "aCompiledMethod parseTree" will miss the compiled method. 
I think it should be set in #parseTree method. This way you will also avoid duplication in reflective method.

Reply:
ah.. I will try that!